### PR TITLE
Use PEM certificates when connecting to KafkaAgent REST endpoint

### PR DESCRIPTION
**Pull request by tinaselenge**
### Type of change
- Enhancement

### Description

This PR is for https://github.com/strimzi/strimzi-kafka-operator/issues/8721. 

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [x] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

**Review by scholzj — COMMENTED** [review]
The code LGTM. I will run the STs. Would it make sense to have some unit tests?

File: operator-common/src/main/java/io/strimzi/operator/common/Util.java [hunk 1]
```diff
@@ -166,6 +166,21 @@ public static byte[] decodeFromSecret(Secret secret, String key) {
         return Base64.getDecoder().decode(secret.getData().get(key));
     }
 
+    /**
+     * Decode binary item that holds Private key in PEM format from Kubernetes Secret from base64 into byte array
```
Comment (see-quick) [inline operator-common/src/main/java/io/strimzi/operator/common/Util.java:170]:
```suggestion
     * Decode the binary item in a Kubernetes Secret, which holds a private key in PEM format, from base64 to a byte array.
```
Just a nit, won't it be better this way?

File: cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaAgentClient.java [hunk 2]
```diff
@@ -98,6 +98,33 @@ private HttpClient createHttpClient() {
         }
     }
 
+    private KeyStore getTrustStore() throws CertificateException, KeyStoreException, IOException, NoSuchAlgorithmException {
+        final CertificateFactory caCertFactory = CertificateFactory.getInstance("X.509");
```
Comment (see-quick) [inline cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/resource/KafkaAgentClient.java:102]: Don't we have some defined Constants for `X.509` or `JKS`? :) If not should we introduce them to eliminate redundancy in the project?

**Review by see-quick — APPROVED** [review]
LGTM, just two nits.

**Review by ppatierno — APPROVED** [review]
LGTM.

File: operator-common/src/main/java/io/strimzi/operator/common/Util.java (hunk 1 above)
Comment (tinaselenge) [reply operator-common/src/main/java/io/strimzi/operator/common/Util.java:170]: yes, that does sound better. Thanks!

**Comment by ppatierno** [comment]
> Would it make sense to have some unit tests?

@tinaselenge are you planning to write some?

**Comment by tinaselenge** [comment]
> > Would it make sense to have some unit tests?
> 
> @tinaselenge are you planning to write some?

I was thinking what would be the best way to test it. I probably would need to create dummy private key/cert and secrets to check if truststore/keystore can be created from the secret? Or would it make more sense to write system test instead that connects to kafka agent with mutual TLS so that we are testing the whole flow?

**Comment by scholzj** [comment]
> Or would it make more sense to write system test instead that connects to kafka agent with mutual TLS so that we are testing the whole flow?

Well, the mTLS is used all the time, so it does no need new STs I think. I guess the `decodePemPrivateKeyFromSecret` would be easy to unit test. But you are right that relying on the ST makes probably sense.

**Review by scholzj — APPROVED** [review]

**Pull request merged**